host-configs/ruby-toss_4_x86_64_ib-clang@16.0.6.cmake

```diff
@@ -0,0 +1,111 @@
+
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+#------------------------------------------------------------------------------
+# Compilers
+#------------------------------------------------------------------------------
+# Compiler Spec: clang@=16.0.6
+#------------------------------------------------------------------------------
+if(DEFINED ENV{SPACK_CC})
+
+  set(CMAKE_C_COMPILER "/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2025_01_31_12_10_06/spack/lib/spack/env/clang/clang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2025_01_31_12_10_06/spack/lib/spack/env/clang/clang++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2025_01_31_12_10_06/spack/lib/spack/env/clang/gfortran" CACHE PATH "")
+
+else()
+
+  set(CMAKE_C_COMPILER "/usr/WS2/smithdev/toss_4_x86_64_ib/llvm/16.0.6/bin/clang" CACHE PATH "")
```

Comment:
Maybe you don't need to double check, but consider verifying all versions match other host configs for consistency.

Reply:
Others should be fine.  This host config is a bit special for now.  It allows us to do CI on Enzyme using a special build of clang and Enzyme hosted in this smithdev directory.  This will probably be integrated better in the future.